src/Plugin/ElasticsearchIndexBase.php

```diff
@@ -110,17 +115,31 @@ public function getClient() {
   }
 
   /**
-   * Gets the event dispatcher.
+   * Returns the event dispatcher.
    *
    * @return \Symfony\Component\EventDispatcher\EventDispatcherInterface
    */
   protected function getEventDispatcher() {
     if (!$this->eventDispatcher) {
       $this->eventDispatcher = \Drupal::service('event_dispatcher');
     }
+
     return $this->eventDispatcher;
   }
 
+  /**
+   * Returns Elasticsearch index plugin manager.
+   *
+   * @return \Drupal\elasticsearch_helper\Plugin\ElasticsearchIndexManager
+   */
+  protected function getElasticsearchIndexPluginManager() {
+    if (!$this->indexPluginManager) {
+      $this->indexPluginManager = \Drupal::service('plugin.manager.elasticsearch_index.processor');
```

Comment:
This plugin could be added through dependency injection?

Reply:
I was trying to avoid adding anything new to the constructor of `ElasticsearchIndexBase` as it might make overall upgrade path a bit messy since a lot of index plugins use their own constructors.